packages/grafana-runtime/src/utils/logging.ts

```diff
@@ -57,3 +57,22 @@ export function logError(err: Error, contexts?: LogContext) {
     });
   }
 }
+
+/**
+ * Wrapper for logX functions.
+ * Creates a logger for frontend monitoring which contains provided "source" and context as defaults when calling logX()
+ */
+export function createMonitoringLogger(source: string, defaultContext?: LogContext) {
+  const createFullContext = (contexts?: LogContext) => ({
+    source: source,
+    ...defaultContext,
+    ...contexts,
+  });
+
+  return {
+    logDebug: (message: string, contexts?: LogContext) => logDebug(message, createFullContext(contexts)),
+    logInfo: (message: string, contexts?: LogContext) => logInfo(message, createFullContext(contexts)),
+    logWarning: (message: string, contexts?: LogContext) => logWarning(message, createFullContext(contexts)),
+    logError: (error: Error, contexts?: LogContext) => logError(error, createFullContext(contexts)),
```

Comment:
Add some comments to get IDE help when using the returned object since the original comments are lost.
```suggestion
    /**
     * Logs a debug message with optional additional context.
     * @param {string} message - The debug message to be logged.
     * @param {LogContext} [contexts] - Optional additional context to be included.
     */
    logDebug: (message: string, contexts?: LogContext) => logDebug(message, createFullContext(contexts)),

    /**
     * Logs an informational message with optional additional context.
     * @param {string} message - The informational message to be logged.
     * @param {LogContext} [contexts] - Optional additional context to be included.
     */
    logInfo: (message: string, contexts?: LogContext) => logInfo(message, createFullContext(contexts)),

    /**
     * Logs a warning message with optional additional context.
     * @param {string} message - The warning message to be logged.
     * @param {LogContext} [contexts] - Optional additional context to be included.
     */
    logWarning: (message: string, contexts?: LogContext) => logWarning(message, createFullContext(contexts)),

    /**
     * Logs an error with optional additional context.
     * @param {Error} error - The error object to be logged.
     * @param {LogContext} [contexts] - Optional additional context to be included.
     */
    logError: (error: Error, contexts?: LogContext) => logError(error, createFullContext(contexts)),
```